File: 1/1000s/17s/1752/CheckIfArrayIsSortedAndRotated.py
# https://leetcode.com/problems/check-if-array-is-sorted-and-rotated/description/?envType=daily-question&envId=2025-02-02

class Solution:
    def check(self, nums: List[int]) -> bool:

        def rotateRight(nums, positions):
            lengthOfNums=len(nums)
            resultantList=[]
            for i in range(0, lengthOfNums, 1):
                resultantList.append(-1)
            for i in range(0, lengthOfNums, 1):
                resultantList[(i+positions)%lengthOfNums]=nums[i]
            return resultantList

        tempNums=[]
        lengthOfNums=len(nums)
        for i in range(0, lengthOfNums, 1):
            tempNums.append(nums[i])
        tempNums.sort()
        for i in range(0, lengthOfNums, 1):
            tocheckNums=rotateRight(nums, i+1)
            if tocheckNums==tempNums:
                return True
        return False
        
        

File: 1/1000s/17s/1752/CheckIfArrayIsSortedAndRotated.cpp
/*
https://leetcode.com/problems/check-if-array-is-sorted-and-rotated/description/?envType=daily-question&envId=2025-02-02
*/

class Solution {
public:

    vector<int> rotateRight(vector<int>&nums, int positions){
        int sz=nums.size();
        vector<int>res(sz, -1);
        for(int i=0; i<sz; ++i){
            res[(i+positions)%sz]=nums[i];
        }
        return res;
    }

    bool check(vector<int>& nums) {
        vector<int>tmpvec=nums;
        sort(tmpvec.begin(), tmpvec.end());
        int sz=nums.size();
        for(int i=0; i<sz; ++i){
            vector<int>tochkvec=rotateRight(nums, i+1);
            if(tochkvec==tmpvec) return true;
        }
        return false;
    }
};

// ------- ******* -------

class SolutionOptimal {
public:
    bool check(vector<int>& nums) {
        int sz=nums.size(), breakCount=0;
        for(int i=0; i<sz; ++i){
            if(nums[i]>nums[(i+1)%sz]) breakCount++;
        }
        return breakCount<=1;
    }
};